standalone/runtime/tests/test_redeem.rs

```diff
@@ -28,117 +28,369 @@ fn consume_to_be_replaced(vault: &mut CoreVaultData, amount_btc: u128) {
+            //  - `amountWrapped` of the redeemer’s tokens MUST be locked by this transaction.
+            //  - If the vault’s collateralization rate is below the PremiumRedeemThreshold, then `redeem.premium` MUST
+            //    be
+            // PremiumRedeemFee multiplied by the worth of `redeem.amountBtc`
+            test_with(|| {
+                let vault_to_be_redeemed = 1500;
+                let user_to_redeem = 1500;
+                set_redeem_state(vault_to_be_redeemed, user_to_redeem, USER, VAULT);
+                setup_redeem(user_to_redeem, USER, VAULT, 1_000_000);
+                let redeem_id = assert_redeem_request_event();
+                let actual_redeem = RedeemPallet::get_open_redeem_request_from_id(&redeem_id).unwrap();
+                assert_eq!(actual_redeem, premium_redeem_request(user_to_redeem, VAULT, USER));
+            });
+        }
+
+        #[test]
+        fn integration_test_redeen_vault_capacity_insufficient() {
+            // Checked PRECONDITION: The vault’s `issuedTokens` MUST be at least `vault.toBeRedeemedTokens +
+            // burnedTokens`.
+            test_with(|| {
+                let vault_to_be_redeemed = 1500;
+                let user_to_redeem = 1500;
+                set_redeem_state(vault_to_be_redeemed, user_to_redeem, USER, VAULT);
+                let core_vault = CoreVaultData::vault(VAULT);
+                CoreVaultData::force_to(
+                    VAULT,
+                    CoreVaultData {
+                        issued: core_vault.issued - 1,
+                        ..core_vault
+                    },
+                );
+                assert_noop!(
+                    Call::Redeem(RedeemCall::request_redeem(
+                        user_to_redeem,
+                        BtcAddress::P2PKH(H160([0u8; 20])),
+                        account_of(VAULT),
+                    ))
+                    .dispatch(origin_of(account_of(ALICE))),
+                    VaultRegistryError::InsufficientTokensCommitted
+                );
+            });
+        }
+
+        #[test]
+        fn integration_test_redeen_dust_value_succeeds() {
+            // Checked PRECONDITION: `burnedTokens` minus the inclusion fee MUST be above the RedeemBtcDustValue,
+            // where the inclusion fee is the multiplication of RedeemTransactionSize and the fee rate estimate
+            // reported by the oracle.
+
+            test_with(|| {
+                // The formula for finding the threshold `to_redeem` for the dust amount error is
+                // `(redeem_dust_value + inclusion_fee) / (1 - redeem_fee_rate)`
+                let redeem_dust_value = RedeemPallet::redeem_btc_dust_value();
+                let inclusion_fee = RedeemPallet::get_current_inclusion_fee().unwrap();
+                let redeem_fee_rate = FeePallet::redeem_fee();
+                let denominator = FixedU128::one().checked_sub(&redeem_fee_rate).expect("Division failed");
```

Comment:
You don't need to use checked math in tests - just use the normal `+`, `-`, `/` etc for readability